packages/react-meteor-data/ReactMeteorData.jsx

```diff
@@ -1,36 +1,54 @@
-const ReactMeteorData = {
-  componentWillMount() {
-    this.data = {};
-    this._meteorDataManager = new MeteorDataManager(this);
-    const newData = this._meteorDataManager.calculateData();
-    this._meteorDataManager.updateData(newData);
-  },
-  componentWillUpdate(nextProps, nextState) {
-    const saveProps = this.props;
-    const saveState = this.state;
-    let newData;
-    try {
-      // Temporarily assign this.state and this.props,
-      // so that they are seen by getMeteorData!
-      // This is a simulation of how the proposed Observe API
-      // for React will work, which calls observe() after
-      // componentWillUpdate and after props and state are
-      // updated, but before render() is called.
-      // See https://github.com/facebook/react/issues/3398.
-      this.props = nextProps;
-      this.state = nextState;
-      newData = this._meteorDataManager.calculateData();
-    } finally {
-      this.props = saveProps;
-      this.state = saveState;
+import React from 'react';
+
+export function connect({ getMeteorData, pure = true }, WrappedComponent) {
```

Comment:
Hi @benjamn, thanks for review. There is no particular reason. Actually, the [convention for `connect`](https://facebook.github.io/react/docs/higher-order-components.html#convention-maximizing-composability) is to be a high-order function that returns a high-component. Like Redux's `connect` and Apollo's `graphql`. You think we should go by that?